src/main/java/com/gw/tasks/GeoweaverProcessTask.java

```diff
@@ -312,12 +312,37 @@ public void startMonitor(javax.websocket.Session socketsession) {
 
   /** This function is called when the task is not loaded by a worker */
   public void endPrematurely() {
-
+    // Mark the task as stopped
     this.curstatus = ExecutionStatus.STOPPED;
 
+    // If the task is running in a separate thread, interrupt the thread
+    if (Thread.currentThread().isInterrupted()) {
```

Comment:
This is very weird and risky way to stop current thread. Please explain why it won't kill geoweaver itself.